File: Client/src/types/addressDetails.ts
export type AddressDetails = {
    id: number;
    unitNumber: string;
    streetNumber: string;
    addressLine1: string;
    addressLine2: string;
    postalCodes: string;
    city: string;
    region: string;
    country: Country;
    isDefault: boolean;
};

type Country = {
    id: number;
    name: string;
    alpha2Code: string;
};

File: API/DTOs/Responses/AddressDetails.cs
using API.Entities;

namespace API.DTOs.Responses;

public class AddressDetails
{
    public int Id { get; set; }
    public string UnitNumber { get; set; } = string.Empty;
    public string StreetNumber { get; set; } = string.Empty;
    public string AddressLine1 { get; set; } = string.Empty;
    public string AddressLine2 { get; set; } = string.Empty;
    public string PostalCodes { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Region { get; set; } = string.Empty;
    public Country Country { get; set; }
    public bool IsDefault { get; set; }
}